# RAT 6.3  

We will use the data in `https://s3-us-west-2.amazonaws.com/dsci/6002/data/homer.csv`.  

In [3]:
%pylab inline

import pandas as pd
import statsmodels.api as sm
import seaborn as sns

homer = pd.read_csv("https://s3-us-west-2.amazonaws.com/dsci/6002/data/homer.csv")

Populating the interactive namespace from numpy and matplotlib


In [4]:
homer.head()

y        x1        x2        x3        x4        x5       x6
0 -2.17360 -0.023816 -0.012885  0.045936 -0.182890  0.028694 -0.81119
1 -1.29570  0.018453  0.058999 -0.072399  0.008679 -0.084344 -0.24208
2  0.14539  0.054999  0.131770 -0.067223  0.202430  0.127060  0.63388
3  0.92959 -0.018933 -0.010875 -0.028255  0.017912  0.084330  0.66062
4  0.90259 -0.000877  0.066941 -0.048190  0.064742  0.147880  0.26643

1) Fit a multiple regression model to predict $y$.  

In [14]:
X = homer[['x1', 'x2', 'x3', 'x4', 'x5', 'x6']]
X = sm.add_constant(X)
y = homer.index

model3 = sm.OLS(y, X)
results3 = model3.fit()
results3.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.001
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                     1.307
Date:                Thu, 06 Oct 2016   Prob (F-statistic):              0.250
Time:                        13:48:31   Log-Likelihood:                -53815.
No. Observations:                6057   AIC:                         1.076e+05
Df Residuals:                    6050   BIC:                         1.077e+05
Df Model:                           6                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [95.0% Conf. Int.]
------------------------------------------------------------------------------
const       3027.8066     22.465    134.777      0.000      2983.767  3071.847
x1          1358.7864    683.662      1.988      0.047        18.565  2699.007
x2            66.7061    419.966      0.159      0.874      -756.576   889.989
x3          -126.3462    626.160     -0.202      0.840     -1353.843  1101.150
x4           -76.2071    222.668     -0.342      0.732      -512.715   360.301
x5          -198.7738    297.788     -0.668      0.504      -782.544   384.996
x6           -48.1169     38.640     -1.245      0.213      -123.865    27.631
==============================================================================
Omnibus:                     5205.221   Durbin-Watson:                   0.003
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              363.529
Skew:                           0.000   Prob(JB):                     1.15e-79
Kurtosis:                       1.800   Cond. No.                         32.1
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

2) Comment on the fit.  

The fit is almost zero. The R^2 is .001

3) Interpret the slope coefficient of $x_1$.  

$y = 22.46 + 1358x_1 + 66.7x_2 -126.34x_3 - 76.2x_4 - 198.77x_5 - 48.12x_6$

4) Is the model useful for predicting $y$? Why or why not?

$no the model is not useful at all. The only variable that seems to be useful is x_1 because its beta's p-value is significant. Everything else just seems to be a bunch of noise.